### DISCUSSION TITLE
How to use nextra on app router under a subdirectory

### DISCUSSION THREAD

**Discussion by ArikSquad** [shuding/nextra#3582] [General]
I have been following the 4.0.0 pre-release branch closely for some time, I've been trying to prepare myself to work with it.

I am having this small problem where I can't seem to make it work fully under `/docs` when I would like to have nextra only under there. I am using `useContentDir: true`. It does successfully show the pages if I load them under the url but the sidebar is completely missing, and the bottom navigation goes to "root" page subdirectories. It doesn't appreciate my `/docs`.

Is this supported or am I doing something wrong? (I'm using latest nextra pre-release & docs-theme)

**Comment by dimaMachina** [comment]
I think you have a similar request as in https://github.com/shuding/nextra/issues/3567 ?

**Reply by ArikSquad** [reply]
It is similar yes but I don't think I get any console errors or logs. But yes, basically the same request. This would be really useful when people want to host a lot of content on their page and have documentation for them without using nextra for everything. I did get that error at some point but it did take tons of iterations (probably because documentation is not done yet). 

I think good support for subdirectories would add a lot. I have something like this on prod using pages router but it is very hacky

**Reply by dimaMachina** [reply]
https://github.com/shuding/nextra/issues/3567#issuecomment-2443981639

wdyt?

**Reply by ArikSquad** [reply]
Seems like good progress!

**Comment by ArikSquad** [comment]
I have tried nextra@4.0.0-app-router.19 but I am experiencing an issue. Here's a screenshot about it:
![image](https://github.com/user-attachments/assets/17bf9395-c6e1-413a-beaa-372b365db762)

Also I would like to clarify my folder setup **(this is most likely unrelated to this issue)**. I have a root layout.tsx that has ThemeProvider and just some small additions, then I have a subfolder (docs) so I can have another layout. Here's a screenshot to visualize it better:
![image](https://github.com/user-attachments/assets/45078550-1ad0-4cd0-85af-2d1399d8dd98)
which is straight from `examples/docs`. The reason why they are subfoldered like this, and not in root layout, is because I want to make it clear that they are used only for docs

**Reply by ArikSquad** [reply]
Oh, nextra appends metadata automatically to every page that doesn't have metadata? And does so even to invalid components?

**Reply by ArikSquad** [reply]
I fixed it by manually removing the mistakes
